src/main/java/org/jabref/Globals.java

```diff
@@ -73,12 +74,16 @@ public static void startBackgroundTasks() {
         Globals.fileUpdateMonitor = new FileUpdateMonitor();
         JabRefExecutorService.INSTANCE.executeInterruptableTask(Globals.fileUpdateMonitor, "FileUpdateMonitor");
 
-        startTelemetryClient();
+        if (Globals.prefs.shouldCollectTelemetry()) {
+            startTelemetryClient();
+        }
     }
 
     private static void stopTelemetryClient() {
-        telemetryClient.trackSessionState(SessionState.End);
-        telemetryClient.flush();
+        if (Globals.prefs.shouldCollectTelemetry()) {
+            getTelemetryClient().ifPresent(client -> client.trackSessionState(SessionState.End));
+            getTelemetryClient().ifPresent(client -> client.flush());
```

Comment:
If you have only a single method call, you can do some syntactic sugar by doing `ifPresent(TelementryClient::flush)`
https://docs.oracle.com/javase/tutorial/java/javaOO/methodreferences.html